run_dir/design/reads_total.html

```diff
@@ -57,25 +57,55 @@ <h3 id="summary_table_section">Summary of Selected {% if "isHiseqX" in readsdata
                     <tr class="darkth">
                         <th><a class="text-decoration-none" href="/project/{{ sample.split('_')[0] }}">{{ sample }}</a></th>
                         <th>%&nbsp;&gt;&nbsp;q30</th>
+                        <th>Mode</th>
+                        <th>Threshold</th>
                         <th>Add</th>
                         <th>{% if "isHiseqX" in readsdata %}Cluster{% else %}Reads{% end %}</th>
                     </tr>
                 {% for d in readsdata[sample] %}
+                    {% set run_setup = d['longer_read_length'] %}
+                    {% try %} {% set run_setup = int(run_setup) %} {% except %}{% end %}
                     <tr class="reads_data">
                         <td class="flowcell"><a class="text-decoration-none" href="/flowcells/{{ d['fcp'].split('_')[0] }}_{{ d['fcp'].split('_')[-1].split(':')[0] }}">{{ d['fcp'] }}</a></td>
                         {% if '_UD' not in d['fcp']  %}
-                            {% if d['q30'] is not None and d['q30']>75 %}
+                            {% if d['run_mode'] == 'HiSeq X' %}
+                                {% set q30_threshold = 75.0 %}
+                            {% elif 'MiSeq' in d['run_mode'] %}
+                                {% if run_setup >= 250 %}
+                                    {% set q30_threshold = 60.0 %}
+                                {% elif run_setup >= 150 %}
+                                    {% set q30_threshold = 70.0 %}
+                                {% elif run_setup >= 100 %}
+                                    {% set q30_threshold = 75.0 %}
+                                {% elif run_setup < 100 %}
+                                    {% set q30_threshold = 80.0 %}
+                                {% end %}
+                            {% else %}
+                                {% if run_setup >= 250 %}
+                                    {% set q30_threshold = 60.0 %}
+                                {% elif run_setup >= 150 %}
+                                    {% set q30_threshold = 75.0 %}
+                                {% elif run_setup >= 100 %}
+                                    {% set q30_threshold = 80.0 %}
+                                {% elif run_setup < 100 %}
+                                    {% set q30_threshold = 85.0 %}
+                                {% end %}
```

Comment:
Move the threshold definition outside the loop perhaps?

Reply:
I changed it a bit now!

Reply:
Ok, that looks cleaner!